docs/quickstart/beginner.rst

```diff
@@ -251,14 +250,26 @@ of an ETL flow.
         row['month'] = date.month
         row['day'] = date.day
 
-Finally, the data can be inserted into the data warehouse. All rows from the
-CSV file are inserted into the location dimension first. This is necessary for
-foreign keys to the location dimension to be computed while filling the fact
-table. The other two dimensions are filled while inserting the facts as the
-data needed is included in the sales records. To ensure that the data is
-committed to the database and that the connection is closed correctly, the
-methods :meth:`.ConnectionWrapper.commit` and :meth:`.ConnectionWrapper.close`
-are executed at the end.
+Finally, the data can be inserted into the data warehouse. With a few
+exceptions, the dimension classes in pygrametl provide the same interface,
+likewise for the fact table classes. When adding members to dimensions or facts
+to fact tables in pygrametl, the methods :meth:`lookup`, :meth:`insert`,
+:meth:`ensure` are generally used. :meth:`lookup` returns the key for the member
+or fact that matches a given row if it exists, :meth:`insert` adds a member to a
+dimension or a fact to a fact table, and :meth:`ensure` performs a
+:meth:`lookup` followed by an :meth:`insert` if the member or fact does not
```

Comment:
would it be easier to understand if "if the member or fact does not exist" was replaced with "if the `lookup` did not find an existing member or fact" or something similar to highlight that this depends on the outcome of `lookup`?

Reply:
Good idea.